chapters/en/module4.md

```diff
@@ -1,110 +1,184 @@
+<slides source='module4/module4_02' shot='0' start='3:42' end='4:35'> </slides>
 </exercise>
 
-<exercise id='4' title='git reset'>
+<exercise id='4' title='Test Your Knowledge'>
+
+You are writing a book using version control. In your local repository you have text file for each chapter. Today, after reading your work again and decided to move some of the text you have written for `chapter3.md` to `chapter1.md`. How would you save this change in Git?
+
+<choice>
+<opt text='It would cut and copy the text into the new file, add both files to the staging area, and then commit with a message explaining the changes.' correct='true'>
+
+</opt>
+
+<opt text='I would <b>revert</b> to the commit I did after writing that paragraph. Then, I would add the text in the correct chapter and make a new commit with a message explaining the changes.'>
+
+Are you sure you want to loose all the changes you did for `chapter2.md`? Even if you can recover those changes, reverting the changes will make you go back to that particular point in the the history of your project.
+
+</opt>
+
+<opt text='I would perform a <b>hard reset</b> to the commit I did after writing that paragraph. Then, I would add the text in the correct chapter and make a new commit with a message explaining the changes.'>
+
+When you realize a hard reset, you are deleting the history of your project until that commit. This means that you will be losing all the changes you did in between that commit.
+
+</opt>
+</choice>
+
+TRUE OR FALSE
 
-What command would you use to move back to X commit for this Y example?
-IN PROGRESS
+Select all the situations when it is a good idea to use reset your work to a previous commit
 
 <choice>
-<opt text='git reset --soft de6aea0' correct='true'>
+<opt text='When you are working in a project locally, without pushing to the remote' correct='true'>
+
+If you are working locally on a private project, it is quite safe to use reset. Nobody is going to be affected by you deleting part of the history of your project.
+
+</opt>
+<opt text='When we are working with other colleagues pushing to the same repository' >
+
+This is not a good idea because you are deleting part of the history of your project. Consider that your colleagues are working with the full history. This could be a problem at the moment of combining the information. In this case probably it is better to revert the changes, a command that we are going to learn in the next sections of this module.
+
 </opt>
-<opt text='git reset --hard de6aea0'>
+
+<opt text='Both are correct' >
+
+You can hard reset your work in both cases, but one option could generate problems.
+
 </opt>
+
 </choice>
 </exercise>
 
-<exercise id='5' title="merge conflicts"  type='slides, video'>
-<slides source='module1/module1_00' shot='0' start='3:42' end='4:35'> </slides>
+<exercise id='5' title='Revert your changes' type='slides, video'>
+<slides source='module4/module4_04' shot='0' start='3:42' end='4:35'> </slides>
```

Comment:
I think this should be `module4/module4_03`; right now I see merge conflict slides when I click on the "Revert your changes" slide deck.

```suggestion
<slides source='module4/module4_03' shot='0' start='3:42' end='4:35'> </slides>
```